src/Oscoin/API/HTTP/Handlers.hs

```diff
@@ -69,26 +69,19 @@ submitTransaction
        , Serialise (Crypto.Signature c)
        , Crypto.HasHashing c
        , Buildable (Crypto.Hash c)
-       , Crypto.HasDigitalSignature c
        )
     => ApiAction c s i a
 submitTransaction = do
-    tx <- getVerifiedTxBody
-    receipt <- node $ do
-        store <- view telemetryStoreL
-        Mempool.addTxs [tx]
-        forM_ [ TxsAddedToMempoolEvent [Crypto.fromHashed $ Crypto.hash @c tx]
-              , TxSubmittedEvent (Crypto.hash @c tx)] $
-            liftIO . Telemetry.emit store
-        pure $ TxSubmitResponse (hash @c tx)
-
-    respond accepted202 $ Ok receipt
-  where
-    getVerifiedTxBody = do
-        tx <- getBody
-        if verifyTx tx
-        then pure tx
-        else respond badRequest400 $ errBody "Invalid transaction signature"
+    tx <- getBody
+    addResult <- node $ Mempool.addTx tx
+    case addResult of
+        Left _ -> respond badRequest400 $ errBody "Invalid transaction signature"
```

Comment:
Would be better if this message just came from the Mempool, since we can't know what the error was.

Reply:
Also, out of loud thought: when we will overhaul our API to be RESTful, we should look at ways to make errors consumable by machines, rather than spitting out an unstructured and unparsable string (but for now, whatever). In the past I have used the [jsend](https://github.com/omniti-labs/jsend) format, but if you know of any better semi-de-facto formats to encode errors, let me know.

Reply:
@cloudhead I agree this is not ideal. I would follow @adinapoli-mndc’s suggestion and think about API errors more holistically at another point. I created an issue for that https://github.com/oscoin/oscoin/issues/558